android/src/com/metalrain/stocksimulator/android/activities/StockSimulatorActivity.java

```diff
@@ -63,12 +69,23 @@ protected void onResume() {
         subscription = StockSimulator.getGameState().bus.toObserverable().subscribe(new Action1<Object>() {
             @Override
             public void call(final Object o) {
+                runOnUiThread(new Runnable() {
```

Comment:
This block of code is duplicated.

If you want the UI to update on other messages, add the messages to the if statement on line 82.